### DISCUSSION TITLE
Migrate to Bun?

### DISCUSSION THREAD

**Discussion by rajdip-b** [keyshade-xyz/keyshade#273] [Ideas]
@kriptonian1, @Sambit003 and anyone else who comes across this issue. Can we integrate bun with keyshade?

**Comment by kriptonian1** [comment]
Yes pls, I would love to take part in the migration process

**Reply by kriptonian1** [reply]
I would recommend going [hono](https://hono.dev/) plus bun because hono is really lightweight, and overall I like there DX. and bun compiles to zig as far as I know, so it will be fast af compared to traditional nodejs and we can also same some cost, but need to check the memory it takes on idle because for nodejs it's a lot.

**Reply by kriptonian1** [reply]
btw I don't think we should considering migrating to hono rn, because then we need to migrate the whole thing

**Reply by kriptonian1** [reply]
okey you know how much time will it take to migrate to bun, just `bun start`

**Reply by rajdip-b** [reply]
0_0

**Reply by kriptonian1** [reply]
we just need to change `node dist/main` to `bun dist/main` LoL

**Reply by Sambit003** [reply]
> we just need to change `node dist/main` to `bun dist/main` LoL

yepp

**Comment by Sambit003** [comment]
we can do one thing, we can move certain parts of the entire application to bun, which actually need faster computation.

**Reply by rajdip-b** [reply]
THE API!

**Reply by Sambit003** [reply]
> THE API!

yepp, we can

**Reply by kriptonian1** [reply]
In that case we can write those parts in rust and do napi, but the question is should this be a concern for as from business perspective.

**Reply by Sambit003** [reply]
> In that case we can write those parts in rust and do napi, but the question is should this be a concern for as from business perspective.

no, then that will be equal to deno, as it's written in rust @kriptonian1

**Reply by Sambit003** [reply]
but deno has out-of-box full and extended support for ts AFAIK @rajdip-b @kriptonian1

**Reply by rajdip-b** [reply]
There are two ways to look at things:

- DevEx
- Performance

And sometimes, both.

But right now, I think we shouldn't focus on performance, since we are not yet dealing with real users. We would need real metrics to understand the performance pain points. Till then, we can just think about DevEx since that matters for a community much more.

So my opinion would be this: move with what seems easy, and has nearly out-of-the-box support, and keep a small window open for improvements.

**Reply by Sambit003** [reply]
that surely sounds good

**Comment by rajdip-b** [comment]
Dumping links:

- https://www.reddit.com/r/nextjs/comments/16hpjs2/is_it_possible_to_migrate_an_existing_nextjs_app/
- https://bun.sh/guides/ecosystem/nextjs
- https://stackoverflow.com/questions/77084231/bun-compatibility-with-nestjs
- https://blog.logrocket.com/migrating-typescript-app-node-js-bun/

**Comment by rajdip-b** [comment]
https://github.com/keyshade-xyz/keyshade/issues/278 Issue is up

**Reply by Sambit003** [reply]
will help you in migrating the API part
